lib/web/apiserver.go

```diff
@@ -2392,17 +2392,62 @@ func (h *Handler) WithClusterAuth(fn ClusterHandler) httprouter.Handle {
 
 type redirectHandlerFunc func(w http.ResponseWriter, r *http.Request, p httprouter.Params) (redirectURL string)
 
+func isValidRedirectURL(redirectURL string) bool {
+	_, err := url.ParseRequestURI(redirectURL)
```

Comment:
lets verify that this redirect starts with a valid scheme `https` to handle the following use-case

```golang
u, err := url.Parse("javascript:alert(1)")
```

which returns true and will result in XSS on the client side

Reply:
At least one of the redirects in the SSO workflow uses regular http (the one that goes to the local redirect server). Is it okay then to also allow http (or maybe check the domain as well when it is http)?